src/lib.rs

```diff
@@ -49,83 +74,36 @@ pub fn arabic2vinculum(input: u64) -> Result<String, String>  {
 /// ```
 pub fn vinculum2arabic<S: AsRef<str>>(input: S) -> Result<u64, String> {
     // input.as_ref()
-    Err("TODO implement".to_string())
+    todo!()
 }
 
-
-fn make_vinculum_number(divisor: u64, times: &u64) -> Result<String, String>
-{
-    match divisor {
-        1000000000 => Ok(make_vinculum_billion(*times)),
-        100000000 => make_vinculum(times, "C̿", "D̿", "M̿"),
-        10000000 => make_vinculum(times, "X̿", "L̿", "C̿"),
-        1000000 => make_vinculum(times, "M̅", "V̿", "X̿"),
-        100000 => make_vinculum(times, "C̅", "D̅", "M̅"),
-        10000 => make_vinculum(times, "X̅", "L̅", "C̅"),
-        1000 => make_vinculum(times, "I̅", "V̅", "X̅"),
-        100 => make_vinculum(times, "C", "D", "I̅"),
-        10 => make_vinculum(times, "X", "L", "C"),
-        1 => make_vinculum(times, "I", "V", "X"),
-        _ => {
-            Err(format!("Unsupported divisor: {}", times))
-        }
+fn make_vinculum_number(power_ten: u32, times: u64) -> Result<String, String> {
+    match CHARACTER_TUPLES.get(&power_ten) {
+        Some(t) => make_vinculum(times, *t),
+        None => Err(format!("Unsupported size: {}", power_ten)),
     }
```

Comment:
Since stuff like this is not expected during runtime and we can guarantee that we'll never panic, something like this is IMHO more appropriate:

```suggestion
    CHARACTER_TUPLES.get(&power_ten).map(|t| make_vinculum(times, *t)).unwrap()
```

Reply:
@aszlig i canNOT promise that I'll never panic LOL